openodia/_understandData.py

```diff
@@ -7,20 +8,40 @@
 
 class UnderstandData:
     """Tokenizer implementation"""
-
+    
     @classmethod
-    def word_tokenizer(cls, text):
+    def word_tokenizer(cls,text):
         """Split the text into words"""
-        # TODO: Do not remove the punctuation characters
+        punct = regex.compile(r'(\p{Punctuation})')    
+        x=punct.split(test_data)
```

Comment:
`test_data` should be `text`

Reply:
Please give the proper variable name depending on what it is and not `x`, `str1` and `i`